core/scripts/chaincli/handler/debug.go

```diff
@@ -228,73 +232,83 @@ func (k *Keeper) Debug(ctx context.Context, args []string) {
-			} else if streamsLookupErr.FeedParamKey != feedIDs || streamsLookupErr.TimeParamKey != timestamp {
+			} else if streamsLookup.IsMercuryV03() {
 				// handle v0.3
-				resolveIneligible("upkeep reverted with StreamsLookup but the configuration is invalid")
-			} else {
 				message("using mercury lookup v0.3")
+			} else {
+				resolveIneligible("upkeep reverted with StreamsLookup but the configuration is invalid")
 			}
-			streamsLookup := &StreamsLookup{streamsLookupErr.FeedParamKey, streamsLookupErr.Feeds, streamsLookupErr.TimeParamKey, streamsLookupErr.Time, streamsLookupErr.ExtraData, upkeepID, blockNum}
 
 			if k.cfg.MercuryLegacyURL == "" || k.cfg.MercuryURL == "" || k.cfg.MercuryID == "" || k.cfg.MercuryKey == "" {
 				failCheckConfig("Mercury configs not set properly, check your MERCURY_LEGACY_URL, MERCURY_URL, MERCURY_ID and MERCURY_KEY", nil)
 			}
-			handler := NewMercuryLookupHandler(&MercuryCredentials{k.cfg.MercuryLegacyURL, k.cfg.MercuryURL, k.cfg.MercuryID, k.cfg.MercuryKey}, k.rpcClient)
-			state, failureReason, values, _, err := handler.doMercuryRequest(ctx, streamsLookup)
-			if failureReason == UpkeepFailureReasonInvalidRevertDataInput {
+
+			// do mercury request
+			automationCheckResult := mustAutomationCheckResult(upkeepID, checkResult, trigger)
+			checkResults := []ocr2keepers.CheckResult{automationCheckResult}
+			values, err := streams.DoMercuryRequest(ctx, streamsLookup, checkResults, 0)
+
+			if automationCheckResult.IneligibilityReason == uint8(mercury.MercuryUpkeepFailureReasonInvalidRevertDataInput) {
 				resolveIneligible("upkeep used invalid revert data")
 			}
-			if state == InvalidMercuryRequest {
+			if automationCheckResult.PipelineExecutionState == uint8(mercury.InvalidMercuryRequest) {
 				resolveIneligible("the mercury request data is invalid")
 			}
 			if err != nil {
-				failCheckConfig("failed to do mercury request ", err)
+				resolveIneligible("failed to DoMercuryRequest")
 			}
-			callbackResult, err := keeperRegistry21.CheckCallback(triggerCallOpts, upkeepID, values, streamsLookup.extraData)
+
+			// do checkCallback
+			err = streams.CheckCallback(ctx, values, streamsLookup, checkResults, 0)
 			if err != nil {
 				failUnknown("failed to execute mercury callback ", err)
 			}
-			if callbackResult.UpkeepFailureReason != 0 {
-				message(fmt.Sprintf("checkCallback failed with UpkeepFailureReason %d", checkResult.UpkeepFailureReason))
+			if automationCheckResult.IneligibilityReason != 0 {
+				message(fmt.Sprintf("checkCallback failed with UpkeepFailureReason %d", automationCheckResult.IneligibilityReason))
 			}
-			upkeepNeeded, performData = callbackResult.UpkeepNeeded, callbackResult.PerformData
-			// do tenderly simulations
-			rawCall, err := core.RegistryABI.Pack("checkCallback", upkeepID, values, streamsLookup.extraData)
+			upkeepNeeded, performData = automationCheckResult.Eligible, automationCheckResult.PerformData
+			// do tenderly simulations for checkCallback
```

Comment:
I hear you! I dont think there is downside of using the streams module, I would say we are closer to the real production env since this debugging script uses the same lib as production. Since we are refactoring the streams module, and the modularized functions come for free. I would just use it. If there is logic gap btw this approach and others, I would consider fixing it then.